### ISSUE TITLE
Quick question: How do I enable threading with the default UI?

### ISSUE THREAD

**Issue by jgadsby** [GetStream/stream-chat-flutter#494]
Just a little confused by this. Another Git issue seems to indicate I need to override the messageBuilder and create my own instance of MessageWidget (along with all the logic for showing different UI elements on the messsage widget) to even get the 'thread reply' button to show up when a message is long-pressed - but that can't be the only way to do this, right? I was hoping to be able to kinda do a MessageWidget.copyWith() where I can override the components I need to tweak or change - or even easier, on the MessageListView, have some sensible and accessible bools for basic features, like 'isThreadingEnabled', 'isPinningEnabled' and so on.

**TL;DR: The defaults are very good and I'm happy to keep them, I just wanted to enable threading in a channel.**

**Comment by rlee1990** [comment]
@jgadsby They have a tutorial here that shows it
[](https://getstream.io/chat/flutter/tutorial/)

**Comment by imtoori** [comment]
as @rlee1990 said you can check the tutorial

We have a property called `threadBuilder` in the `MessageListView` that you can use for returning your `ThreadPage` widget

In the `ThreadPage` you can use the `MessageListView.parentMessage` param to pass the `parentMessage` that you get 
from the `threadBuilder` to use the `MessageListView` in "thread-mode"

Also the `MessageInput` accept a `parentMessage` to send messages to that specific thread
